_test/test_sqw/test_PixelData_operations.m

```diff
@@ -87,31 +95,172 @@ function test_compute_bin_data_correct_output_5_pages_mex_1_thread(obj)
+        unary_ops = {
+            @acos, [0, 1], ...
+            @acosh, [1, 3], ...
+            @acot, [0, 1], ...
+            @acoth, [10, 15], ...
+            @acsc, [1, 3], ...
+            @acsch, [1, 3], ...
+            @asec, [1.5, 3], ...
+            @asech, [0, 1], ...
+            @asin, [0, 1], ...
+            @asinh, [1, 3], ...
+            @atan, [0, 1], ...
+            @atanh, [0, 0.5], ...
+            @cos, [0, 1], ...
+            @cosh, [0, 1], ...
+            @cot, [0, 1], ...
+            @coth, [1.5, 3], ...
+            @csc, [0.5, 2.5], ...
+            @csch, [1, 3], ...
+            @exp, [0, 1], ...
+            @log, [1, 3], ...
+            @log10, [1, 3], ...
+            @sec, [2, 4], ...
+            @sech, [0, 1.4], ...
+            @sin, [0, 3], ...
+            @sinh, [0, 3], ...
+            @sqrt, [0, 3], ...
+            @tan, [0, 1], ...
+            @tanh [0, 3], ...
+        };
+
+        % For each unary operator, perform the operation on some file-backed
+        % data and compare the result to the same operation used on the same
+        % data all held in memory
+        num_pix = 7;
+        npix_in_page = 3;
+        for i = 1:numel(unary_ops)/2
+            unary_op = unary_ops{2*i - 1};
+            data_range = unary_ops{2*i};
+
+            data = obj.get_random_data_in_range(obj.NUM_COLS_IN_PIX_BLOCK, ...
+                                                num_pix, data_range);
+            pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+            pix = pix.do_unary_op(unary_op);
+
+            transformed_data = obj.concatenate_pixel_pages(pix);
+
+            pix_in_mem = PixelData(data);
+            pix_in_mem = pix_in_mem.do_unary_op(unary_op);
+            expected_transformed_data = pix_in_mem.data;
+
+            assertEqual(transformed_data, expected_transformed_data, ...
+                        sprintf('Failure calculating ''%s''', char(unary_op)), ...
+                        obj.FLOAT_TOLERANCE);
+        end
     end
 
+    function test_do_unary_op_returns_correct_output_with_cosine_1_page(obj)
+        data = rand(9, 50);
+        npix_in_page = 50;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        pix = pix.do_unary_op(@cos);
+
+        original_signal = data(obj.SIGNAL_IDX, :);
+        original_variance = data(obj.VARIANCE_IDX, :);
+
+        expected_data = data;
+        expected_data(obj.SIGNAL_IDX, :) = cos(original_signal);
+        expected_data(obj.VARIANCE_IDX, :) = ...
+                abs(1 - pix.signal.^2).*original_variance;
+
+        assertEqual(pix.data, expected_data);
+    end
+
+    % -- Helpers --
+    function pix = get_pix_with_fake_faccess(obj, data, npix_in_page)
+        faccess = FakeFAccess(data);
+        mem_alloc = npix_in_page*obj.NUM_BYTES_IN_VALUE*obj.NUM_COLS_IN_PIX_BLOCK;
+        pix = PixelData(faccess, mem_alloc);
+    end
+
+    function data = concatenate_pixel_pages(obj, pix)
```

Comment:
does this function need testing? Its correct behaviour is critical to the all-operator test above.